modules/pref/src/main/NotificationPref.scala

```diff
@@ -0,0 +1,118 @@
+package lila.pref
+
+import play.api.libs.json.{ Json, OWrites }
+import reactivemongo.api.bson.*
+import NotificationPref.*
+
+case class Allows(value: Int) extends AnyVal with IntValue:
+  def push: Boolean   = (value & NotificationPref.PUSH) != 0
+  def web: Boolean    = (value & NotificationPref.WEB) != 0
+  def device: Boolean = (value & NotificationPref.DEVICE) != 0
+  def bell: Boolean   = (value & NotificationPref.BELL) != 0
+  def any: Boolean    = value != 0
+
+object Allows:
+  val canFilter = Set(
+    "privateMessage",
+    "challenge",
+    "mention",
+    "streamStart",
+    "tournamentSoon",
+    "gameEvent",
+    "invitedStudy"
+  )
+
+  def fromForm(bell: Boolean, push: Boolean): Allows =
+    Allows((bell ?? BELL) | (push ?? PUSH))
+
+  def toForm(allows: Allows): Some[(Boolean, Boolean)] =
+    Some((allows.bell, allows.push))
+
+  def fromCode(code: Int) = Allows(code)
+
+case class NotifyAllows(userId: UserId, allows: Allows):
+  def push   = allows push
+  def web    = allows web
+  def device = allows device
+  def bell   = allows bell
+  def any    = allows any
+
+// take care with NotificationPref field names - they map directly to db and ws channels
+
+case class NotificationPref(
+    privateMessage: Allows,
+    challenge: Allows,
+    mention: Allows,
+    streamStart: Allows,
+    tournamentSoon: Allows,
+    gameEvent: Allows,
+    invitedStudy: Allows,
+    correspondenceEmail: Int
```

Comment:
why is this an `Int` and not a `Boolean`?

Reply:
Probably carried over from the view layer, which itself was just an Int to maintain uniformity with the rest of the boolean prefs that are treated as ints.  But at the model layer you're right it should be a bool